src/scripts/post_process_externally_managed_content.py

```diff
@@ -0,0 +1,112 @@
+import click
+import os
+import pandas as pd
+import logging
+
+logger = logging.getLogger(__name__)
+
+def is_qc_failure_related_to_external_source(qc_failure, erroneous_row, external):
```

Comment:
```suggestion
def is_qc_failure_related_to_external_source(qc_failure, erroneous_row, external) -> Union[bool, str]:
```

### Function name & typing: `is_qc_failure_related_to_external_source()`
I see this is related to the cool system you guys have put together for reporting QC failures in `mondo-ingest` as mentioned on [slack](https://monarchinitiative.slack.com/archives/G01JACCEDAB/p1724105022538769).

Just noting that this was confusing to me. Because of `is_` at the beginning of the function name, I at first expected this to only return a boolean.

I'm suggesting `Union[bool, str]` return type, but you can also make it just `str` by returning `''` in the case of no matches instead of `False`.

Really though this function should be renamed, but I'm not yet sure what to.

Reply:
I agree with the function name - it used to be a boolen method, but it evolved. Now I changed it to `get_column_of_external_source_related_to_qc_failure`. I didnt add your annotations because if we go this way we need to add them in 100 places - no time!